libsolidity/experimental/analysis/TypeInference.cpp

```diff
@@ -128,6 +128,12 @@ bool TypeInference::analyze(SourceUnit const& _sourceUnit)
 	return !m_errorReporter.hasErrors();
 }
 
+bool TypeInference::visit(ForAllQuantifier const& /*_forAllQuantifier*/)
+{
+	// Nothing to do here. All the magic happens when we visit the parameter list.
```

Comment:
That doesn't seem quite right to me - wouldn't it still need to switch to type context? I.e. won't it parse the variable declaration in the parameter list as *terms* now instead of as types?

Reply:
In effect the main difference would be if ``forall a:C`` treats ``C`` as type or as class...

Reply:
Yeah, you're probably right. Might be why I'm getting this error: https://github.com/ethereum/solidity/pull/14668#discussion_r1383516823.

Reply:
Yep, I was just about to comment that there :-).

Reply:
Fixed.